exercises/practice/binary-search-tree/BinarySearchTree.cs

```diff
@@ -1,53 +1,21 @@
 using System;
-using System.Collections;
 using System.Collections.Generic;
+using Newtonsoft.Json;
 
-public class BinarySearchTree : IEnumerable<int>
+public class BinarySearchTree<T> where T : IComparable
 {
-    public BinarySearchTree(int value)
-    {
-    }
+    public int Count
+    => throw new NotImplementedException("Please implement the Count property of the BinarySearchTree<T> class.");
```

Comment:
This style is more commonly used in the code base:

```suggestion
    public int Count =>
        throw new NotImplementedException("Please implement the Count property of the BinarySearchTree<T> class.");
```